### DISCUSSION TITLE
How to import external library in boot file?

### DISCUSSION THREAD

**Discussion by elie464** [quasarframework/quasar#10453] [General - Components / Directives / etc] [answered]
I wrote a boot file `chart.js` like this to import from Chartjs

```
import { boot } from 'quasar/wrappers';
import Chart from 'chart.js/auto';

export default boot(async ({ app }) => {
  app.use('Chart', Chart);
})
```

Then included `chart` in quasar.conf `boot: ['chart']`

When i call it in a script like `new Chart(ctx, { type: 'bar', data: this.chartData, options: this.options });` in a component I made it doesn't work saying Chart is not defined. However if I import directly into the component file i.e. the `import Chart from 'chart.js/auto';` line it works. How can i properly write a boot file so i don't have to import it directly in a component and globally define it?

TIA

**Comment by MatthewSH** [comment, marked as answer]
I think you have a fundamental misunderstanding of what boot files are for and what they're not for.
Quote from [Boot Files - Example of unneeded usage of boot files](https://quasar.dev/quasar-cli/boot-files#example-of-unneeded-usage-of-boot-files) 
> For plain JavaScript libraries like Lodash, which don’t need any initialization prior to their usage. Lodash, for example, might make sense to use as a boot file only if you want to inject Vue prototype with it, like being able to use this.$_ inside your Vue files.

Some things to note:
1. You're attempting to call `app.use` on something that's not built for Vue so it has no idea what to do with it, unless ChartJS is doing some magic that automatically detects this? Still not sure.
2. You're attempting to `use` an entire library...still not a good thing in this case.
3. You should definitely read up on boot files and learn the appropriate usage according to Quasar team.
4. The error you get is absolutely correct. Chart is undefined because you're not even attempting to use it correctly, based on the code given, even if `app.use` worked. 

I'd recommend importing it as needed and if you use it consistently enough where you constantly reimport it and create new charts, then a) make a reusable component to render the charts or b) create some sort of chart service file that imports and returns a new chart for you

**Reply by elie464** [reply]
ok thanks

**Reply by MatthewSH** [reply]
No problem! Let me know if there's anything else! 😃
